runtime/common/MeasureCounts.cpp

```diff
@@ -165,8 +165,9 @@ sample_result::sample_result(std::vector<ExecutionResult> &results) {
   for (auto &result : results) {
     sampleResults.insert({result.registerName, result});
   }
-  for (auto &[bits, count] : results[0].counts)
-    totalShots += count;
+  if (!results.empty())
```

Comment:
Q: Should this check be done at the beginning and perhaps throw exception if `results.empty()` ?

Reply:
> Should this check be done at the beginning?

I think it is not necessary since the above `for` loops should be handling empty vectors just fine.  

> perhaps throw exception if `results.empty()`

I think it's better to graciously handle an empty vector as if it was the default constructor rather than throwing in this case.